src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs

```diff
@@ -250,6 +253,7 @@ internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert,
             }
 
             jsonTypeInfo.OnDeserialized?.Invoke(obj);
+            jsonTypeInfo.CheckRequiredProperties(ref state);
```

Comment:
Same as above, possibly with a more descriptive name:
```C#
state.Current.ValidateAllRequiredPropertiesAreRead(jsonTypeInfo);
```

Reply:
I'm missing the "above" comment but will fix this